src/withStyles.jsx

```diff
@@ -67,7 +70,23 @@ export function withStyles(
             }}
           />
         );
-      }
+      };
+    } else {
+      // WrappedComponent is a function, so we can avoid a little overhead by
+      // calling it directly instead of rendering it. This avoids React
+      // lifecycle overhead for the wrapped component and may not be necessary
+      // in a future version of React.
+      WithStyles.prototype.render = function render() {
+        const { themeName } = this.context;
+        flushIfNecessary();
+        return WrappedComponent({
```

Comment:
I'm really really skeptical that this is a good idea; it seems like something that would be very brittle, and could easily break in React itself. What overhead is there exactly? Shouldn't this be something React optimizes?

What about PropType checks? I realize these are hoisted, but custom propTypes might not function the same when hoisting, so I think it's still important to run them on the inner component.

Reply:
This has been sitting for a little, but I wanted to bring up this discussion if you were not aware of it:  https://github.com/acdlite/recompose/pull/538 Recompose pulled this type of optimization out due to how brittle things were in production builds vs development.  Andrew Clark comments towards the bottom of the PR on why it is not currently in React,